base/build-static-ioc.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+set -ex
+
+IOC=$1
+
+cp /opt/epics/RELEASE configure/
+cat << EOF > configure/CONFIG_SITE.local
+STATIC_BUILD=YES
+FULL_STATIC_BUILD=YES
+USR_LDFLAGS = -static-pie
+EOF
```

Comment:
I feel like overwriting this can be an issue for some IOCs. I'm okay with it, but it seems worth documenting it.

Reply:
`CONFIG_SITE.local` shouldn't be checked into git. What sort of IOCs do you think this could be an issue for?

Reply:
Indeed. That doesn't mean people follow this, though. For instance, [`ether_ip` is a module that does not][1]... there are certainly [others around][2].

But my point is that someone can attempt to build any IOC, and suddenly all her well-crafted and not versioned configuration in`CONFIG_SITE.local` will be overwritten in her host machine, since it is volume mounted. She won't be able to `git restore` it, because it shouldn't even be tracked by Git in the first place.

Just to be clear, I don't see overwriting an issue for us, since we won't be using any local site configuration for both RFFE and AFC IOCs. But future users might have to use them for overwriting default settings they care about.

[1]: https://github.com/epics-modules/ether_ip/blob/6ca0e66ef72510afb51b4dfab987b0ef5d9b8d36/configure/CONFIG_SITE.local
[2]: https://github.com/search?q=path%3Aconfigure%2FCONFIG_SITE.local&type=Code&ref=advsearch&l=&l=

Reply:
This is somewhat related to an annoyance I have had with the RELEASE file being marked as changed in git...

Do you think it would be a good alternative to do the following?

1. backup `RELEASE` and `CONFIG_SITE.local`
2. add our own versions, and use them for build
3. restore them when the script exits and with a trap handler, in case one of the make commands errors out

Reply:
What do you think of this solution?

Reply:
> What do you think of this solution?

I think we are almost there. It misses one thing: we should append to the end of existing `CONFIG_SITE.local`, and not create it entirely from scratch. After all, if someone maintains a local `CONFIG_SITE`, they have something they want to (or have to) customize in the build.

Reply:
In a similar vein, I'd consider appending (`+=`) to `LD_FLAGS` instead of overwriting (`=`) it.

Reply:
> > What do you think of this solution?
> 
> I think we are almost there. It misses one thing: we should append to the end of existing `CONFIG_SITE.local`, and not create it entirely from scratch. After all, if someone maintains a local `CONFIG_SITE`, they have something they want to (or have to) customize in the build.

I don't know how much I agree with this :thinking: 

The opposite situation is a custom `CONFIG_SITE.local` for building outside of the container, and using that erroneously when you mean to build using just the options from the script.

> In a similar vein, I'd consider appending (+=) to LD_FLAGS instead of overwriting (=) it.

That's fair

Reply:
> The opposite situation is a custom CONFIG_SITE.local for building outside of the container, and using that erroneously when you mean to build using just the options from the script.

Indeed. Not sure the best way to deal with this. In this case, the user has better control of the situation, I think. She can rename the file to preserve her custom out-of-container settings, and let it clean for the script. In the situation I had described, she could change the Git-tracked `CONFIG_SITE`, or write her own version of `lnls-build-static-ioc`.

None of them seems optimal in the end. I think the latter is messier as it involves changing a tracked file, or rewriting something non-trivial.

If you prefer, we can let this as is, and change this if someone gets annoyed by it in the future.

Reply:
Add `-include CONFIG_SITE.local.lnls-build-static-ioc` to the `CONFIG_SITE.local` I write?

Reply:
> > The opposite situation is a custom CONFIG_SITE.local for building outside of the container, and using that erroneously when you mean to build using just the options from the script.
> 
> Indeed. Not sure the best way to deal with this. In this case, the user has better control of the situation, I think. She can rename the file to preserve her custom out-of-container settings, and let it clean for the script. In the situation I had described, she could change the Git-tracked `CONFIG_SITE`, or write her own version of `lnls-build-static-ioc`.
> 
> None of them seems optimal in the end. I think the latter is messier as it involves changing a tracked file, or rewriting something non-trivial.
> 
> If you prefer, we can let this as is, and change this if someone gets annoyed by it in the future.

Regardless of the solution used, I think that all these decisions that the user has to make must be very well documented it README.md to prevent any erroneously use of this feature

Reply:
> Add -include CONFIG_SITE.local.lnls-build-static-ioc to the CONFIG_SITE.local I write?

That seems a good option if we continue down this road.

> Regardless of the solution used, I think that all these decisions that the user has to make must be very well documented it README.md to prevent any erroneously use of this feature

Agreed.